Add common protection api support & small fixes

EvieTheOwl opened the pull request:
Added support for Patbox's common protection API (https://github.com/Patbox/common-protection-api) which prevents explosions bypassing claims for land claiming mods like Cadmus (https://modrinth.com/mod/cadmus)

Also fixes #44 and an issues where you could get floating fire blocks on the client

Commit:
Add common protection api support

Commit:
Update version

Pyrofab commented on src/main/java/ladysnake/blast/common/block/BonesburrierBlock.java:
```diff
@@ -43,9 +43,12 @@ public static void prime(World world, BlockPos pos) {
-            entity.setPos(pos.getX() + 0.5, pos.getY(), pos.getZ() + 0.5);
-            world.spawnEntity(entity);
-            world.playSound(null, entity.getX(), entity.getY(), entity.getZ(), SoundEvents.ENTITY_TNT_PRIMED, SoundCategory.BLOCKS, 1.0F, 1.0F);
+            if (entity != null) {
```
Should probably error here if it is in fact null

Pyrofab reviewed (commented):
Will review this more in depth later but at first glance it's good work again - just makes me think we should really refactor the block classes at some point to reduce the code duplication

Commit:
Fix Bonesburrier protection

Commit:
Remove null checks and make switch to interact claim check for knockback explosion

EvieTheOwl commented:
> Will review this more in depth later but at first glance it's good work again - just makes me think we should really refactor the block classes at some point to reduce the code duplication

True. That's gonna take a bit longer I think. Will of course be worth switching to variables that mean something as well

EvieTheOwl commented:
Video of it working: https://youtu.be/6LrdmD8crsg

Commit:
Switch to canExplode instead of canBreak

Commit:
Small fixes, including #44

- Prevent floating fire blocks being displayed on the client
- Add entity language translations

sleepy-evelyn commented:
Now fixes #44 & a bug where fire could be seen floating on the client after exploding gunpowder

Commit:
Update the wiki article

Commit:
Reduce header size

Commit:
Simplify header title

EvieTheOwl commented:
Added documentation to the readme and sent a PR for the wiki: https://github.com/Ladysnake/Ladysnake.github.io/pull/77.
Unless any issues or changes are required should be finished with this PR now

Pyrofab commented on src/main/java/ladysnake/blast/common/Blast.java:
```diff
@@ -1,14 +1,19 @@
 import net.minecraft.network.PacketByteBuf;
+import net.minecraft.server.network.ServerPlayerEntity;
 import net.minecraft.util.math.Direction;
+import org.jetbrains.annotations.Nullable;
```
Are these imports used anywhere ? I do not see any change elsewhere in the file

Pyrofab commented on src/main/java/ladysnake/blast/common/entity/ShrapnelBlockEntity.java:
```diff
@@ -0,0 +1,78 @@
+    @Override
+    protected void writeCustomDataToNbt(NbtCompound nbt) {
+        super.writeCustomDataToNbt(nbt);
+        if (!getWorld().isClient && owner != null)
```
Could you add braces for those 2-lines statements ? It's a little thing but I think it improves homogeneity and prevents the stupidest kind of bugs.

Pyrofab commented on src/main/java/ladysnake/blast/common/block/StripminerBlock.java:
```diff
@@ -44,10 +44,12 @@ public static void primeStripminer(World world, BlockPos pos) {
             entity.setPos(pos.getX() + 0.5, pos.getY(), pos.getZ() + 0.5);
             world.spawnEntity(entity);
             world.playSound(null, entity.getX(), entity.getY(), entity.getZ(), SoundEvents.ENTITY_TNT_PRIMED, SoundCategory.BLOCKS, 1.0F, 1.0F);
+
```
Why the extra space ?

Pyrofab reviewed (commented):
Not easy to review with all the existing decompiled code, but looks good to me. Just had to annoy you with some code style things, which by the way reminded me to add an editorconfig to the project.

Commit:
Optimise all imports for the project

Commit:
Fixed formatting issues

Pyrofab reviewed (approved):
Very cool thanks
